srml/bridge/src/lib.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Bridge Module
+//!
+//! This will eventually have some useful documentation.
+//! For now though, enjoy this cow's wisdom.
+//!
+//! ________________________________________
+//! / You are only young once, but you can  \
+//! \ stay immature indefinitely.           /
+//! ----------------------------------------
+//!        \   ^__^
+//!         \  (oo)\_______
+//!            (__)\       )\/\
+//!                ||----w |
+//!                ||     ||
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use sr_primitives::{
+	generic::BlockId,
+	traits::{Block as BlockT, Header as HeaderT},
+};
+use support::{decl_error, decl_event, decl_module, decl_storage, dispatch::Result};
+use system::{ensure_root, ensure_signed};
+
+// struct Bridge {}
+//
+// impl Bridge {
+// 	pub fn new() -> Self {}
+// 	pub fn submit_claim() -> Result<(), Err> {}
+// 	pub fn get_id(&self) -> u64 {}
+// 	pub fn get_recently_finalized_block(&self) -> Block {}
+// }
+
+pub trait Trait: system::Trait + session::Trait {
+	// The identifier type for an authority.
+	// type AuthorityId: Member + Parameter + RuntimeAppPublic + Default;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Bridge {
+		/// Get the ID of the current bridge
+		pub BridgeId get(bridge_id): T::Hash;
+
+		/// Get the ID of the current bridge
+		// TODO: Figure out how to ge a Block here
+		pub LastFinalizedBlock get(last_finalized_block): u64; // Block;
```

Comment:
is this any different from `system::block_number()`?

Reply:
It looks like that returns the current block number, which isn't necessarily what we want here. See my comment above.

Reply:
ahh so it is the `LastFinalizedBlock ` of some other chain? Then I'd guess it could be named a bit better + store like a tuple of this finalize hash and some identifier of the chain if needed.